core/src/main/scala/org/apache/spark/shuffle/sort/SortShuffleManager.scala

```diff
@@ -79,6 +83,8 @@ private[spark] class SortShuffleManager(conf: SparkConf) extends ShuffleManager
    */
   private[this] val numMapsForShuffle = new ConcurrentHashMap[Int, Int]()
 
+  private lazy val shuffleExecutorComponents = loadShuffleExecutorComponents(conf)
+
   override val shuffleBlockResolver = new IndexShuffleBlockResolver(conf)
```

Comment:
Should we remove this hardcoded `IndexShuffleBlockResolver`? Due to it's only used in `DefaultShuffleDataIO`. And this `IndexShuffleBlockResolver` has already(as it should be, dealt with in certain implementation) be constructed in https://github.com/apache/spark/blob/2421c928dd35ccaa28825a96bcd3449e94383b59/core/src/main/java/org/apache/spark/shuffle/sort/io/DefaultShuffleExecutorComponents.java#L43

Reply:
Currently the `SortShuffleManager.blockResolver` is called in `BlockManager`, probably we should modify that too.

Reply:
I really want to avoid touching code as much as possible actually. So while it might currently be more semantically correct to do this refactor, in practice in the interest of reducing risk for performing extraneous refactors, would like to defer on this. We can add a `TODO` to flag this.

Reply:
Will this TODO be resolved in later PRs of 25299?

Reply:
I'm not sure if these series of PRs are the right place to do such refactors.

Reply:
`SortShuffleManager` is for all ShuffleIO implementations after your series, but `IndexShffleBlockResolver` is only for the default local implementation(vanilla Spark's), so...

Reply:
Yes, please add a TODO and create a JIRA to track this.